DEVELOPMENT.md

```diff
@@ -230,14 +230,18 @@ of:
 - **If you change a package's deps** (including adding an external dependency),
   then you must run [`./hack/update-deps.sh`](./hack/update-deps.sh).
 
-These are both idempotent, and we expect that running these at `HEAD` to have no
+- **If you change the schema for a custom resource** (for example, adding fields), then you must update
+  the relevant section of [`./hack/schemapatch-config.yaml`](./hack/schemapatch-config.yaml)
+  and run [`./hack/update-schemas.sh`](./hack/update-schemas.sh).
```

Comment:
This is not "quite" true: You must only update the schemapatch-config file, if you're changing what's allowed to pass via the PodSpec. Other schema changes to our types (i.e. adding fields) will automatically be picked up without any config.

While we clarify that, we could also add a remark about different use-cases maybe, as-in:

1. If a new field gets added without any feature-gating, add it to the `allowedFields` list.
2. If a new field gets added behind a feature flag, **don't** add it to `allowedFields` (as we don't want to advertise it unconditionally) but rather make sure that it's parent type has `preserveUnknownFields: true # for feature flagged fields` set on it, so we don't prune the field.